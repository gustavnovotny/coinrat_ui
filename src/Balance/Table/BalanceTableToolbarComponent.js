// @flow
import React from "react"
import {Checkbox, Label} from "@blueprintjs/core"
import {Box, Flex} from "reflexbox"

type Props = {
  hideZeroBalances: boolean,
  onChangeShowZeroBalances: (showZeroBalances: boolean) => void,
}

const BalanceTableToolbarComponent = ({hideZeroBalances, onChangeShowZeroBalances}: Props) => {
  return <Flex>
    <Box><Label text="Hide zero balances:"/></Box>
    <Box><Checkbox style={{marginLeft: 4 + 'px'}} checked={hideZeroBalances} onChange={onChangeShowZeroBalances}/></Box>
  </Flex>
}

export default BalanceTableToolbarComponent
