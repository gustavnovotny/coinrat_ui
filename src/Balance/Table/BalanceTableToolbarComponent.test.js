// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import BalanceTableToolbarComponent from "./BalanceTableToolbarComponent"

it('renders correctly, hide zero balances ON', () => {
  const onChangeShowZeroBalances = () => undefined
  const tree = renderer
    .create(<BalanceTableToolbarComponent hideZeroBalances={true} onChangeShowZeroBalances={onChangeShowZeroBalances}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly, hide zero balances OFF', () => {
  const onChangeShowZeroBalances = () => undefined
  const tree = renderer
    .create(<BalanceTableToolbarComponent hideZeroBalances={false} onChangeShowZeroBalances={onChangeShowZeroBalances}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
