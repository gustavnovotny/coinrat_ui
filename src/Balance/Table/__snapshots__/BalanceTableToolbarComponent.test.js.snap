// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly, hide zero balances OFF 1`] = `
<div
  className="_Rfx0_"
>
  <div
    className=""
  >
    <div
      className="pt-label"
    >
      Hide zero balances:
      <span
        className="pt-text-muted"
      />
    </div>
  </div>
  <div
    className=""
  >
    <label
      className="pt-control pt-checkbox"
      style={
        Object {
          "marginLeft": "4px",
        }
      }
    >
      <input
        checked={false}
        onChange={[Function]}
        style={
          Object {
            "marginLeft": "4px",
          }
        }
        type="checkbox"
      />
      <span
        className="pt-control-indicator"
      />
    </label>
  </div>
</div>
`;

exports[`renders correctly, hide zero balances ON 1`] = `
<div
  className="_Rfx0_"
>
  <div
    className=""
  >
    <div
      className="pt-label"
    >
      Hide zero balances:
      <span
        className="pt-text-muted"
      />
    </div>
  </div>
  <div
    className=""
  >
    <label
      className="pt-control pt-checkbox"
      style={
        Object {
          "marginLeft": "4px",
        }
      }
    >
      <input
        checked={true}
        onChange={[Function]}
        style={
          Object {
            "marginLeft": "4px",
          }
        }
        type="checkbox"
      />
      <span
        className="pt-control-indicator"
      />
    </label>
  </div>
</div>
`;
